components/scream/src/control/tests/sc_tests.cpp

```diff
@@ -0,0 +1,127 @@
+#include "share/grid/point_grid.hpp"
+#include "control/surface_coupling.hpp"
+
+#include <ekat/util/ekat_test_utils.hpp>
+
+#include <catch2/catch.hpp>
+
+TEST_CASE ("surface_coupling")
+{
+  // Some namespaces/aliases
+  using namespace scream;
+  using namespace ShortFieldTagsNames;
+  using namespace ekat::units;
+  using FL = FieldLayout;
+  using FID = FieldIdentifier;
+  using RPDF = std::uniform_real_distribution<Real>;
+  using rngAlg = std::mt19937_64;
+
+  // Some constants
+  constexpr int ncols = 4;
+  constexpr int nlevs = 8;
+  constexpr int nruns = 10;
+
+  // Create a comm
+  ekat::Comm comm (MPI_COMM_WORLD);
+
+  // The random numbers generator
+  std::random_device rd;
+  const unsigned int catchRngSeed = Catch::rngSeed();
+  const unsigned int seed = catchRngSeed==0 ? rd() : catchRngSeed;
+  if (comm.am_i_root()) {
+    std::cout << "seed: " << seed << (catchRngSeed==0 ? " (catch rng seed was 0)\n" : "\n");
+  }
+  rngAlg engine(seed);
+  RPDF pdf(0.0,1.0);
+
+  // Create a grid
+  auto grid = create_point_grid("my grid",ncols*comm.size(), nlevs, comm);
+
+  // Create some field ids, and register them in a field repo
+  // Note: we create two repos, so we can compare outputs with inputs
+  FID s2d_id("s2d",FL{{COL},{ncols}},Pa,grid->name());
+  FID s3d_id("s3d",FL{{COL,VL},{ncols,nlevs}},Pa,grid->name());
+  FID v3d_id("v3d",FL{{COL,CMP,VL},{ncols,2,nlevs}},m/s,grid->name());
+
+  FieldRepository<Real> repo_in, repo_out;
+  repo_in.registration_begins();
+  repo_in.register_field(s2d_id);
+  repo_in.register_field(s3d_id);
+  repo_in.register_field(v3d_id);
+  repo_in.registration_ends();
+  repo_out.registration_begins();
+  repo_out.register_field(s2d_id);
+  repo_out.register_field(s3d_id);
+  repo_out.register_field(v3d_id);
+  repo_out.registration_ends();
+
+  // Create two SC objects, to import and export
+  control::SurfaceCoupling importer(grid,repo_in);
+  control::SurfaceCoupling exporter(grid,repo_out);
+
+  importer.set_num_fields(4,0); // Recall that SC counts *scalar* fields, so vector3d counts as 2 fields
+  exporter.set_num_fields(0,4);
+
+  // Register fields in the importer/exporter
+  importer.register_import("s2d",0);
+  importer.register_import("s3d",1);
+  importer.register_import("v3d",2,0);
+  importer.register_import("v3d",3,1);
+  exporter.register_export("s2d",0);
+  exporter.register_export("s3d",1);
+  exporter.register_export("v3d",2,0);
+  exporter.register_export("v3d",3,1);
+
+  // Create a raw array big enough to contain all the 2d data for import/export
+  Real* raw_data = new Real[ncols*4];
```

Comment:
is the 4 written here related to the total number of variables?  If so, then the `int` variable suggested in the earlier comment could be used here too.